pkg/prometheus/promcfg.go

```diff
@@ -1871,12 +1876,45 @@ func (cg *ConfigGenerator) generateRemoteWriteConfig(
 		}
 
 		if spec.AzureAD != nil {
-			azureAd := yaml.MapSlice{
-				{
-					Key: "managed_identity", Value: yaml.MapSlice{
+			if spec.AzureAD.ManagedIdentity.ClientID == "" && spec.AzureAD.OAuth == nil {
+				return yaml.MapItem{}, fmt.Errorf("must provide Azure Managed Identity or Azure OAuth in the Azure AD config")
+			}
+			if spec.AzureAD.ManagedIdentity.ClientID != "" && spec.AzureAD.OAuth != nil {
+				return yaml.MapItem{}, fmt.Errorf("cannot provide both Azure Managed Identity and Azure OAuth in the Azure AD config")
+			}
```

Comment:
I wonder if there isn't a more appropriate place for the validation? The method signature is only generateRemoteWriteConfig and doesn't mention any kind of validation. I also don't see other remote write options being validated here.

Sorry for giving such a vague answer for this one, at the moment I can't remember where exactly we validate remote-write, but I have a feeling we do it elsewhere 😅

Reply:
by validation I mean, we don't return errors here because we've already checked it beforehand

Reply:
Thanks in general for all the feedback! Specifically this comment -> I've moved it to the `ValidateRemoteWriteSpec` function and reverted the signature of the `generateRemoteWriteConfig` to no error returns. Makes sense.